### PR TITLE
build: update notice to 2023

### PR DESCRIPTION
**Pull request by kruskall**
It's 2023 :tada: 

CI is failing because the year is not up to date.

### CODE REVIEW

**Review by marclop — APPROVED** [review]

File: NOTICE.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Copyright 2022 Elasticsearch BV
+Copyright 2023 Elasticsearch BV
```
Comment (marclop) [inline NOTICE.txt:1]: Shouldn't it be `2022-2023`?

```suggestion
Copyright 2022-2023 Elasticsearch BV
```

Comment (kruskall) [reply NOTICE.txt:1]: Nope, the template says `currentYear`: https://github.com/elastic/apm-aws-lambda/blob/main/scripts/templates/NOTICE.txt.tmpl#L13

Comment (kruskall) [reply NOTICE.txt:1]: I'm not sure what we're doing in other repositories. It might be worth to look into it and make it more consistent

Comment (marclop) [reply NOTICE.txt:1]: The APM Server used that format, that's why I was suggesting we use that format here as well.

Comment (kruskall) [reply NOTICE.txt:1]: I've looked at a few repositories and I think the lambda extension is the only one using `currentYear`. I'll gather more data and open a followup issue/PR.

**Pull request merged**